### PR TITLE
Initial work on Travis CI

### PR DESCRIPTION
**Pull request by TrechNex**
This will be a branch a number of us collaborate on before we merge it.

### CODE REVIEW

**Comment by TrechNex** [comment]
My current thinking is that this is good enough to sanity check code changes against a build.

I think Jenkins is better placed to do the final Linux & Windows packaging. That also affords us the opportunity to beta test the builds before formally releasing and attaching to a release tag.

**Comment by TrechNex** [comment]
Before merging the next step would be to fix the credentials so they work for all branches that might be created for this repository.

**Comment by BrainBlasted** [comment]
For the Linux CI, what if we built inside the flatpak container instead of the distro?

**Comment by BrainBlasted** [comment]
See here for an example travis.yml: https://github.com/writeas/writeas-gtk/blob/ce92eed6cf8cf1f5c336aa6928fa96af17c52d1c/.travis.yml

**Comment by TrechNex** [comment]
Mostly because it'd take longer than 20 minutes and would need to be published somewhere for you to make use of it :)

This was why I proposed doing packaged builds on Jenkins and just a basic sanity build on Travis for merge requests. The latter is more practical if it needs to run on every new commit in a pull request.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+- "./autogen.sh --prefix=$INSTALL_PREFIX"
+- "./configure --prefix=$INSTALL_PREFIX"
+- make && make install
+- cd $SRC_PREFIX/libmypaint
+- "./autogen.sh --prefix=$INSTALL_PREFIX"
+- "./configure --prefix=$INSTALL_PREFIX"
+- make && make install
+- cd $SRC_PREFIX/gegl
+- "./autogen.sh --prefix=$INSTALL_PREFIX"
+- make && make install
```
Comment (bochecha) [inline .travis.yml]: I would do these as `before_script` personally.

Not very important, more a matter of taste I guess. :slightly_smiling_face:

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    - libopenjp2-7-dev
+    - libmng-dev
+    - libheif-dev
+    - libwmf-dev
+    - libxpm-dev
+    - xvfb
+    - gnome-icon-theme
+    - llvm
+    - autopoint
+    - flatpak
```
Comment (bochecha) [inline .travis.yml]: I don't think you need `flatpak` here?

File: .travis.yml (hunk 1 above)
Comment (bochecha) [inline .travis.yml]: A few of these deps (at least the ones hosted on gnome.org) should have a `--disable-gtk-doc` switch or something similar.

It avoids building the docs, which saves some time.

Comment (TrechNex) [reply .travis.yml]: Done for all submodules :)

File: .travis.yml (hunk 2 above)
Comment (TrechNex) [reply .travis.yml]: Removed in latest commit

File: .travis.yml (hunk 1 above)
Comment (bochecha) [reply .travis.yml]: So for babl and gegl, the option is `--disable-docs`.

Comment (TrechNex) [reply .travis.yml]: Good catch! I had to dig through the wiki archives to find the answer: https://wiki.gimp.org/index.php?title=Hacking:Building/Windows&oldid=1819

Comment (bochecha) [reply .travis.yml]: Heh, I just went and read the 4 `configure.ac` files, that seemed quicker. :stuck_out_tongue:

**Pull request merged**